bindings/nodejs/src/lib.rs

```diff
@@ -50,66 +78,189 @@ pub struct ObjectMeta {
 }
 
 #[napi]
-pub struct Operator {
-    inner: opendal::Operator
+pub enum ObjectMode {
+    /// FILE means the object has data to read.
+    FILE,
+    /// DIR means the object can be listed.
+    DIR,
+    /// Unknown means we don't know what we can do on this object.
+    Unknown,
 }
 
+#[allow(dead_code)]
 #[napi]
-impl Operator {
-    pub fn new(op: opendal::Operator) -> Self {
+pub struct ObjectMetadata {
+    /// Mode of this object.
+    pub mode: ObjectMode,
+
+    /// Content-Disposition of this object
+    pub content_disposition: Option<String>,
+    /// Content Length of this object
+    pub content_length: Option<u32>,
+    /// Content MD5 of this object.
+    pub content_md5: Option<String>,
+    /// Content Range of this object.
+    pub content_range: Option<Vec<u32>>,
+    /// Content Type of this object.
+    pub content_type: Option<String>,
+    /// ETag of this object.
+    pub etag: Option<String>,
+    /// Last Modified of this object.
+    pub last_modified: i64,
+}
+
+#[napi]
+pub struct ObjectLister {
+    inner: opendal::ObjectLister
+}
+
+#[napi]
+impl ObjectLister {
+    pub fn new(op: opendal::ObjectLister) -> Self {
         Self { inner: op }
     }
 
     #[napi]
-    pub async fn meta(&self, path: String) -> Result<ObjectMeta> {
-        let o = self.inner.object(&path);
-        let meta = o
+    pub async unsafe fn next_page(&mut self) -> Result<Vec<DataObject>> {
```

Comment:
We don't need to export this API in bindings. Users can use `ObjectLister` as an async iterator.

Reply:
I need to verify it.

Reply:
Can't make a js native iterator directly.
Return an object with a `next` method as an alternative.